autofs-5.0.5 - fix sasl bind host name selection

From: Ian Kent <ikent@redhat.com>

A host name is needed when starting a sasl client. But the ldap
library can return a list of host names with the port appended
following a colon. With IPv6 the host name itself may have colons
so a forward search can no longer be used to find the colon
delimiting the port number and a reverse search can't be used
because it may end up including more than one host name.
---

 CHANGELOG            |    1 +
 modules/cyrus-sasl.c |    7 +++++++
 2 files changed, 8 insertions(+)


--- autofs-5.0.5.orig/CHANGELOG
+++ autofs-5.0.5/CHANGELOG
@@ -58,6 +58,7 @@
 - add option to dump configured automount maps.
 - reset negative status on cache prune.
 - fix wildcard map entry match.
+- fix sasl bind host name selection.
 
 03/09/2009 autofs-5.0.5
 -----------------------
--- autofs-5.0.5.orig/modules/cyrus-sasl.c
+++ autofs-5.0.5/modules/cyrus-sasl.c
@@ -738,6 +738,13 @@ sasl_bind_mech(unsigned logopt, LDAP *ld
 		return NULL;
 	}
 
+	/*
+	 * We need a host name to start the client.
+	 * But the ldap library can return a list of host names so
+	 * just use the first one.
+	 */
+	if ((tmp = strchr(host, ' ')))
+		*tmp = '\0';
 	if ((tmp = strrchr(host, ':'))) {
 		if (*(tmp - 1) != ']') {
 			*tmp = '\0';
